src/main/java/seedu/address/logic/commands/RankCommand.java

```diff
@@ -0,0 +1,152 @@
+        }
+        Predicate<Participation> filterByCompetition = p -> p.getCompetition().isSameElement(competition);
+        model.updateFilteredParticipationList(filterByCompetition);
+        List<Participation> participationList = model.getFilteredParticipationList();
+        List<Participation> pLists = copy(participationList);
+        pLists.sort(new ParticipationComparator());
+        int rank = 1;
+        int score = 0;
+        for (Participation p : pLists) {
+            if (p.getPerson().isSameElement(athlete)) {
+                score += p.getTotalScore();
+                break;
+            }
+            rank += 1;
+        }
+
+        return new CommandResult(String.format(MESSAGE_SUCCESS, rank, score, athleteName, competitionName));
+    }
+
+    /**
+     * Copies an obs list into a normal java list for sorting purposes
+     *
+     */
+    private List<Participation> copy (List<Participation> participationList) {
+        List<Participation> pLists = new ArrayList<>();
+        for (Participation p : participationList) {
+            pLists.add(p);
+        }
+        return pLists;
+
+    }
+
+    /**
+     * Filters for competition with competitionName
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return Competition object if competition with competitionName exists, else null is returned.
+     */
+    private Competition getSpecificComp(Model model) {
+        List<Competition> competitionList = model.getFilteredCompetitionList();
+        Competition competition = null;
+        for (Competition c : competitionList) {
+            if (c.getName().equals(this.competitionName)) {
+                competition = c;
+                break;
+            }
+        }
+        return competition;
+
+    }
+
+    /**
+     * Filters for athlete with athleteName
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return Person object if a Participation contains person with athleteName and competition with competitionName
+     */
+    private Person getSpecificAthlete(Model model) {
+        List<Participation> participationList = model.getFilteredParticipationList();
+        Person athlete = null;
+        for (Participation p : participationList) {
+            boolean isSameAthleteName = p.getPerson().getName().equals(this.athleteName);
+            boolean isSameCompName = p.getCompetition().getName().equals(this.competitionName);
+            if (isSameAthleteName && isSameCompName) {
+                athlete = p.getPerson();
+                break;
+            }
+        }
+        return athlete;
+    }
+
+    /**
+     * Compares two participations for sorting
+     *
+     */
+    public static class ParticipationComparator implements Comparator<Participation> {
+
+        @Override
+        public int compare(Participation p1, Participation p2) {
+            return p2.getTotalScore() - p1.getTotalScore();
+
+        }
+    }
```

Comment:
i think its p1.getTotalScore - p2.getTotalScore

also,

can rename the comparator to participationTotalScoreComparator in case we need to add other comparators